src/main/java/org/apache/maven/shared/utils/io/FileUtils.java

```diff
@@ -161,47 +161,52 @@ protected FileUtils()
      * Returns the directory path portion of a file specification string.
      * Matches the equally named unix command.
      *
-     * @param filename the file path
+     * @param path the file path
      * @return the directory portion excluding the ending file separator
+     * @deprecated use {@code Paths.get(path).getParent().getName()}
      */
-    @Nonnull public static String dirname( @Nonnull String filename )
+    @Deprecated
+    @Nonnull public static String dirname( @Nonnull String path )
```

Comment:
This hunk is especially great because from a filename you cannot derive a dirname.